storage/interceptors/broker_public_plans_interceptor.go

```diff
@@ -53,7 +56,8 @@ func (p *PublicPlanCreateInterceptorProvider) Name() string {
 
 type PublicPlanUpdateInterceptorProvider struct {
 	IsCatalogPlanPublicFunc publicPlanProcessor
-	SupportedPlatforms      func(ctx context.Context, plan *types.ServicePlan, repository storage.Repository) ([]string, error)
+	SupportedPlatforms      func(ctx context.Context, plan *types.ServicePlan, repository storage.Repository) (map[string]*types.Platform, error)
```

Comment:
don't you want to change the other occurrences to `SupportedPlatformsProcessor` as well?